app/src/main/java/com/concordium/wallet/ui/walletconnect/WalletConnectService.kt

```diff
@@ -165,6 +175,23 @@ class WalletConnectService : Service(), SignClient.WalletDelegate, CoreClient.Co
         }
     }
 
+    private fun matchesURLScheme(wcUri: String): Boolean =
+        Regex("wc|concordiumwallet://wc").containsMatchIn(wcUri)
+
+    private fun getConnectionParams(wcUri: String): String {
+        return when {
+            wcUri.startsWith("wc:") -> wcUri
+            else -> {
+                try {
+                    Uri.parse(wcUri)
+                        .getQueryParameters("uri").first()
+                } catch (e: RuntimeException) {
+                    ""
```

Comment:
I think it is a bit hacky to return an empty result, and then we have to know that is an error case. I am not sure how it is generally done in the Android wallet, but perhaps you could use https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/.

Just try to avoid being implicit by having an empty string mean that it failed.